src/java-opentimelineio/src/main/java/io/opentimeline/opentimelineio/Composition.java

```diff
@@ -149,6 +150,37 @@ public void setChildren(List<Composable> children, ErrorStatus errorStatus) {
 
     public native HashMap<Composable, TimeRange> getRangeOfAllChildren(ErrorStatus errorStatus);
 
+    public <T extends Composable> Stream<T> eachChild(
+            TimeRange searchRange, Class<T> descendedFrom, boolean shallowSearch, ErrorStatus errorStatus) {
+        List<Composable> children;
+        if (searchRange != null) {
```

Comment:
`if (searchRange != null)` does not look like it makes any difference to have searchRange. Possibly this will be sorted after?

Reply:
Ohhh. I'm sorry. I think I forgot about this completely. I need to implement that part. Thanks for pointing it out.

Reply:
@margant I think you might've seen an older commit. I actually made a change in a later commit. If you choose the option to see commits from all changes you might be able to see it.

Reply:
https://github.com/PixarAnimationStudios/OpenTimelineIO/pull/753/commits/559930e5481d47e2d146be45d0a1183f9ecd6b55

Here's the commit